swift/Arrow/Tests/ArrowTests/CodableTests.swift

```diff
@@ -0,0 +1,168 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import XCTest
+@testable import Arrow
+
+final class CodableTests: XCTestCase {
+    public class TestClass: Codable {
+        public var prop0: Bool
+        public var prop1: Int8
+        public var prop2: Int16
+        public var prop3: Int32
+        public var prop4: Int64
+        public var prop5: UInt8
+        public var prop6: UInt16
+        public var prop7: UInt32
+        public var prop8: UInt64
+        public var prop9: Float
+        public var prop10: Double
+        public var prop11: String
+        public var prop12: Date
+
+        public required init() {
+            self.prop0 = false
+            self.prop1 = 1
+            self.prop2 = 2
+            self.prop3 = 3
+            self.prop4 = 4
+            self.prop5 = 5
+            self.prop6 = 6
+            self.prop7 = 7
+            self.prop8 = 8
+            self.prop9 = 9
+            self.prop10 = 10
+            self.prop11 = "11"
+            self.prop12 = Date.now
```

Comment:
Using type name instead of `propN` may be easier to understand. For example: `prop0` -> `bool` and `prop1` -> `int8`